### ISSUE TITLE
Add examples for using basic auth and custom headers

### ISSUE THREAD

**Issue by jordansissel** [elastic/go-elasticsearch#31]
While pairing with @Dr-Syn on using this package, we found that basic auth seems to be not yet documented?

For basic auth:
* Scanning the code gives the feeling that including the user:pass in the `Addresses` list works for this purpose

For custom headers
* It's also useful sometimes to add custom headers, for example, with es-security-runas-user or for passing other credential headers (JWT tokens, Cookies, etc).

**Comment by tty2** [comment]
1) Library uses `net/http` so you can see how to set basic auth in it's documentation.
2) Why don't you set needed headers in your application side?

**Comment by karmi** [comment]
I've added the configuration options for HTTP Basic Authentication in the attached commit.

When it comes to custom headers, the correct solution in the overall spirit of the package is to provide a custom transport, following eg. the example in [`_examples/customization.go`](https://github.com/elastic/go-elasticsearch/blob/master/_examples/customization.go); I'll update the example a bit, so it's more illustrative.

Sidenote: In the prototyping stage, I've experimented with exposing a function, `OnRequest(r *http.Request)`, to allow total control over request and response manipulation. In the end, the general consensus was to _not_ add it, because it added too much weight to the core code, and could lead to unpredictable and hard to debug problems. Therefore, the focus is on making it easy for people to add custom transport, whenever they need to manipulate the request and response in any way.
